src/objectExplorer/objectExplorerService.ts

```diff
@@ -43,10 +58,46 @@ export class ObjectExplorerService {
         const handler = (result: SessionCreatedParameters) => {
             if (result.success) {
                 let nodeLabel = this._nodePathToNodeLabelMap.get(result.rootNode.nodePath);
-                self._currentNode = TreeNodeInfo.fromNodeInfo(result.rootNode, result.sessionId, self.currentNode, nodeLabel);
-                self._rootTreeNodeArray.push(self.currentNode);
+                // if no node label, check if it has a name in saved profiles
+                // in case this call came from new query
```

Comment:
Shouldn't you be checking if nodeLabel is undefined at this point so you don't have to do the logic if we already have a label?